### PR TITLE
feat: added `v3` API version for the detector `captaindatago`

### LINKED ISSUE #3483: Introduce new `v3` version for the detector `captaindata`
Please review the [Community Note](https://github.com/trufflesecurity/trufflehog/blob/main/.github/community_note.md) before submitting

### Expected Behavior

Add a new `v3` detector for `captaindata`.

Screenshot 2024-10-20 at 1 04 07 AM

Reference : https://support.captaindata.co/hc/en-us/articles/6065714910877-API-endpoints-documentation-reference
API v3 : https://docs.captaindata.co/#1da39f5f-a26f-4a53-9e84-3e4b28077eae
## Environment
 * OS: [e.g. iOS]
 * Version [e.g. 22]

### References

* #0000

### PR DESCRIPTION
**Pull request by sahil9001**
### Description:
Fixes #3483 

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

**Review by rgmz — CHANGES_REQUESTED** [review]
Same comment as https://github.com/trufflesecurity/trufflehog/pull/3414#discussion_r1801705144 and https://github.com/trufflesecurity/trufflehog/pull/3427#discussion_r1803346687

**Comment by sahil9001** [comment]
Verified with API Key @rgmz 
Screenshot 2024-10-22 at 2 01 09 AM

**Review by rgmz — APPROVED** [review]

File: pkg/detectors/captaindata/v3/captaindata.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
```
Comment (kashifkhan0771) [inline pkg/detectors/captaindata/v3/captaindata.go:21]: The versioning interface is currently unimplemented. When running trufflehog, the following warning is displayed:
```
2024-10-25T11:39:18+05:00	info-0	trufflehog	possible duplicate detector configured	{"detector": "CaptainData"}
```

**Review by kashifkhan0771 — CHANGES_REQUESTED** [review]

File: pkg/detectors/captaindata/v3/captaindata.go (hunk 1 above)
Comment (sahil9001) [reply pkg/detectors/captaindata/v3/captaindata.go:21]: How are you testing this @kashifkhan0771 , can you tell me the command?

Comment (kashifkhan0771) [reply pkg/detectors/captaindata/v3/captaindata.go:21]: Add your token in some file locally and run following command against it
```
trufflehog filesystem ~/<path_to_file_where_secret_is>
```

Comment (sahil9001) [reply pkg/detectors/captaindata/v3/captaindata.go:21]: Thanks, I checked with the changes, this warning is not coming now, I have fixed it.

File: pkg/engine/defaults.go [hunk 2]
```diff
@@ -1642,6 +1642,8 @@ func DefaultDetectors() []detectors.Detector {
 		saladcloudapikey.Scanner{},
 		boxoauth.Scanner{},
 		apimetrics.Scanner{},
+		captainDataV2.Scanner{},
```
Comment (kashifkhan0771) [inline pkg/engine/defaults.go:1645]: Where is `V1`?

**Review by kashifkhan0771 — CHANGES_REQUESTED** [review]

File: pkg/engine/defaults.go (hunk 2 above)
Comment (kashifkhan0771) [reply pkg/engine/defaults.go:1645]: If there is no `V1` , We should start with `V1`

Comment (sahil9001) [reply pkg/engine/defaults.go:1645]: > Where is `V1`?

I named them according to API versions available, there was `v2` and `v3` hence the name.

Comment (kashifkhan0771) [reply pkg/engine/defaults.go:1645]: I believe the versioning is based on detectors not the API. Correct me If I am wrong @zricethezav @abmussani 
Correct way would be to start with V1

Comment (sahil9001) [reply pkg/engine/defaults.go:1646]: Done, please check @kashifkhan0771

**Review by kashifkhan0771 — APPROVED** [review]
LGTM! Good Work @sahil9001 - Thanks for the contribution.

**Pull request merged**